add new Accelerator plugin

galpt opened the pull request:
add new plugin called 'Accelerator' to improve discord client performance

galpt committed:
add new Accelerator plugin

add new plugin called 'Accelerator' to improve discord client performance

verticalsync commented:
Please move all your files to `src/equicordplugins/`, and put yourself into EquicordDevs not vencord Devs

galpt commented:
ahh, I thought equicordplugins is for only the official dev team. okay will do

galpt committed:
Revert "add new Accelerator plugin"

This reverts commit 16c92129f02d556a07f5398beec9d942b65cdf7a.

galpt committed:
moved accelerator dir to equicordplugins dir

added a new 'accelerator' plugin and moved the dir to 'equicordplugins' dir (previously it's in 'plugins' dir)

KrstlSkll69 commented:
Put yourself near the bottom of the equicorddevs section not at the tol

galpt commented:
okay

galpt committed:
Update constants.ts

galpt committed:
Revert "Update constants.ts"

This reverts commit 638cbcb1ed7c776bd70fb94bb22a97f1029748d7.

galpt committed:
Revert "moved accelerator dir to equicordplugins dir"

This reverts commit 95dc23b6a26a092174309708772d16f3d251dff4.

galpt committed:
revision

galpt committed:
Add smart request scheduler and improve preloading

Introduces a smart request scheduler to manage and pace network requests, reducing congestion and improving user experience. Adds an experimental setting to enable request scheduling, integrates the scheduler with message preloading, and enhances the stats module's uptime formatting for better readability.

The scheduler is inspired from queue management like FQ_CODEL and CAKE and the principles can actually be implemented in application layer. Although javascript might be slower compared to the original implementations written in C.
Some references:
1. https://datatracker.ietf.org/doc/html/rfc8290
2. https://man7.org/linux/man-pages/man8/tc-cake.8.html

thororen1234 commented:
I don't really accept prs with DOM manipulation anymore, observers, etc. Also wouldn't OpenASAR be a better alterative?

galpt commented:
OpenAsar mostly works on optimizing Discord's frontend. Users are welcome to use OpenAsar if they prefer it over Equicord's existing features, or they can simply use Equicord if they want a straightforward solution without external additions.

My quick look at the OpenAsar repository suggests it doesn't manage backend components like the `scheduler.ts` module, which was modeled after Linux's qdisc.

I understand your point, and I'll close this PR.
